java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodCallExprContext.java

```diff
@@ -148,51 +149,14 @@ private MethodUsage resolveMethodTypeParametersFromExplicitList(TypeSolver typeS
 
     @Override
     public SymbolReference<ResolvedMethodDeclaration> solveMethod(String name, List<ResolvedType> argumentsTypes, boolean staticOnly, TypeSolver typeSolver) {
-        if (wrappedNode.getScope().isPresent()) {
-            Expression scope = wrappedNode.getScope().get();
-
-            // consider static methods
-            if (scope instanceof NameExpr) {
-                NameExpr scopeAsName = (NameExpr) scope;
-                SymbolReference<ResolvedTypeDeclaration> symbolReference = this.solveType(scopeAsName.getName().getId(), typeSolver);
-                if (symbolReference.isSolved() && symbolReference.getCorrespondingDeclaration().isType()) {
-                    ResolvedTypeDeclaration typeDeclaration = symbolReference.getCorrespondingDeclaration().asType();
-                    return MethodResolutionLogic.solveMethodInType(typeDeclaration, name, argumentsTypes, false, typeSolver);
-                }
-            }
-
-            ResolvedType typeOfScope = null;
-            try {
-                typeOfScope = JavaParserFacade.get(typeSolver).getType(scope);
-            } catch (Exception e) {
-                throw new RuntimeException(String.format("Issue calculating the type of the scope of " + this), e);
+        Collection<ResolvedReferenceTypeDeclaration> rrtds = findTypeDeclarations(wrappedNode.getScope(), typeSolver);
+        for (ResolvedReferenceTypeDeclaration rrtd : rrtds) {
+            SymbolReference<ResolvedMethodDeclaration> res = MethodResolutionLogic.solveMethodInType(rrtd, name, argumentsTypes, false, typeSolver);
```

Comment:
Do we need to consider the types in a specific order? What if multiple types give an answer?